### ISSUE TITLE
Problem using treeWAS with phylogenetic tree

### ISSUE THREAD

**Issue by VincentGreffe** [caitiecollins/treeWAS#8]
Dear Caitie, 

I would love to try out treeWAS, but I'm currently struggling with a problem using it with my own phylogenetic tree.
If I use my phenotype and snps data as input only (and tree = 'NJ'), it works perfectly fine but if I add my own phylogenetic tree I get following error: 

"Error in dimnames(x) <- dn : 
  length of 'dimnames' [1] not equal to array extent"

I used the read.tree function from the ape package, and everything seems fine (labels match those of snps and phen, class = phylo). 
Is it possible to help me further with this problem? 

Kind regards,

Vincent

**Comment by caitiecollins** [comment]
Hi Vincent, 

Happy to see you're trying out treeWAS. 

Sorry about the issue you've run into, especially because I would generally recommend that users use their own trees where possible (your basic NJ tree is only so reliable). 

I'm afraid the error message you received is not very specific. I'm not sure where to begin, seeing as I've not seen that one come out of treeWAS before. 

Is there any way you might be able to send me at least the tree you're working with? If you can send all of your data without risking any privacy issues, that would be even more helpful, though as you believe it is the tree causing the issue, that would be a good start. Would that be possible?

Best, 
Caitlin.

**Comment by caitiecollins** [comment]
Hi again, 

Bit of a stab in the dark, but I went looking for any places in treeWAS where rownames are assigned, and I've made one change that could conceivably prevent the type of error you are running into. 

Without knowing more about the data/tree you are working with, unfortunately I've no idea if this is actually the solution to your problem. That said, if it's not difficult, you're welcome to re-download and install the now-updated version of treeWAS and see if the error returns when repeating your analysis. Otherwise, I will wait to hopefully get a look at your tree and then go from there. 

Cheers, 
CC

**Comment by VincentGreffe** [comment]
Thank you for your quick response! 
I downloaded the updated version and redid the analysis, but unfortunately it gave the same results: again it worked without any tree, but when I added my own tree it gave an error.
However now I got a different error: 

```
 Error in get.ancestral.pars(var = snps, tree = tree, unique.cols = TRUE) : 
   tree$tip.label and rownames(snps)
                                                      must contain the same set of labels
                                                       so that individuals can be correctly identified.
```

I checked if all the tree tip labels occur in the rownames of the snps and vice versa (as suggested in the vignette) and it seemed fine.

My tree was constructed with RAxML (to avoid privacy issues I adjusted the names of the tip labels):

> ((A:0.00130719070539194230,(((B:0.00001650670500063382,C:0.00000851131048370177):0.00082337970946691789,((D:0.00000532527046838464,E:0.00000100000050002909):0.00008226438498828017,(F:0.00100682803992606720,G:0.00007259976954739289):0.00023337574388439269):0.00110678330074304274):0.00048982432726672807,((G:0.00069994418467503729,H:0.00059560398415553864):0.00037491763120454420,(I:0.00135295743202488495,((J:0.00000100000050002909,(K:0.00000453280289915745,((L:0.00000741724886627452,M:0.00000100000050002909):0.00000100000050002909,N:0.00000100000050002909):0.00000100000050002909):0.00000100000050002909):0.00288488551516075976,(O:0.00170064709951211319,(P:0.00092096423865302012,Q:0.00071623633593621220):0.00096247447923951206):0.00199433245422529120):0.00100280677512064797):0.00107388424516662116):0.00041491974866823789):0.00107024082964636016):0.00104299888739011609,((R:0.00135348835502869646,(S:0.00124659557151733983,(T:0.00070586070076103047,(U:0.00084493354894050164,V:0.00064772691520446923):0.00025731775330368280):0.00049694969484594272):0.00062115925055869199):0.00050824605784040040,(((W:0.00100314059531373431,X:0.00120425146630555985):0.00041000663804046153,Y:0.00215574266597259000):0.00017730180865793046,((Z:0.00000133415548067811,AA:0.00000213176637023572):0.00092765635859488578,((AB:0.00000239371696009947,AC:0.00000373339072057526):0.00086433381862764517,(AD:0.00124652776706772541,((((AE:0.00000106241924206686,AF:0.00000100000050002909):0.00082193157897600009,AG:0.00105319431899062593):0.00053546813780252990,(AH:0.00070641172612912243,AI:0.00073055693613807043):0.00014143840903263079):0.00482135925040616878,(AJ:0.00165263007300070356,(AK:0.00076872202903174901,((AL:0.00000931171211502609,AM:0.00000292550193295960):0.00000184776161230403,AN:0.00000374766647137600):0.00137946701770354338):0.00124616077126835176):0.00142467294995195898):0.00157969567374186383):0.00039404015388974238):0.00067145141172096027):0.00036733143295338358):0.00057943635158902180):0.00030456231226200697,AO:0.00130438243307226857):0.0;

**Comment by caitiecollins** [comment]
Boom. Solved. (Or at least I'm pretty sure it should be.)

There was an issue with how it handled non-binary trees (it was getting confused by the three-way split occurring at the root of your tree). But I believe I've managed to resolve that now. Thank you for sending me your tree, by the way. I wouldn't have found the issue without it. 

Ok, here's hoping everything works smoothly now. Let me know how it goes? 

Best, 
Caitlin.
